specs/OAuth/AuthManager.md

```diff
@@ -0,0 +1,891 @@
+AuthManager API
+===
+
+This is the spec for proposal: [Issue #441](https://github.com/microsoft/WindowsAppSDK/issues/441)
+
+This spec details the API of a new `AuthManager` in WinAppSDK,
+enabling desktop applications such as WinUI3 to seamlessly perform OAuth functionality across diverse Windows platforms.
+
+
+
+# Background
+
+The current OS 
+[WebAuthenticationBroker](https://docs.microsoft.com/uwp/api/windows.security.authentication.web.webauthenticationbroker),
+primarily designed for UWP applications,
+presents several challenges when used in desktop environments. 
+Key issues include the dependency on `ApplicationView`, 
+which are not compatible with desktop app frameworks.
+As a result, developers are forced to resort to workarounds involving interop interfaces and 
+additional code to implement OAuth 2.0 functionality into their WinUI 3 desktop applications.
+
+In response to this feedback, this `AuthManager` API for WinAppSDK aims to provide 
+a streamlined solution that meets the expectations of our developer community. 
+The goal is to offer seamless OAuth 2.0 capabilities with full feature parity across all Windows platforms supported by WinAppSDK. 
+This new API eliminates the need for cumbersome workarounds and 
+simplifies the process of incorporating OAuth 2.0 functionality into desktop applications.
+
+
+This new WinAppSDK version of the `AuthManager` is different than the existing public
+[WebAuthenticationBroker](https://learn.microsoft.com/en-us/uwp/api/windows.security.authentication.web.webauthenticationbroker?view=winrt-22621) 
+API. We've opted to follow OAuth best practices more closely - e.g. using user's default browser. The best practices for the API are taken from the
+IETF(Internet Engineering Task Force) OAuth 2.0 Authorization Framework [RFC 6749](https://tools.ietf.org/html/rfc6749), 
+Proof Key for Code Exchange (PKCE) [RFC 7636](https://tools.ietf.org/html/rfc7636),
+and OAuth 2.0 for Native Apps [RFC 8252](https://tools.ietf.org/html/rfc8252).
+
+
+
+# Conceptual pages
+
+ ## Perform OAuth 2.0 (c#)
+
+ Performing an Authorization Code Request (grant type/'response_type' = "code")
+
+ ```c#
+AuthRequestParams authRequestParams = AuthRequestParams::CreateForAuthorizationCodeRequest(L"my_client_id",
+    Uri(L"my-app:/oauth-callback/"));
+authRequestParams.Scope(L"user:email user:birthday");
+
+AuthRequestResult authRequestResult = co_await AuthManager::InitiateAuthRequestAsync(
+    Uri(L"https://my.server.com/oauth/authorize"), authRequestParams);
+if (AuthResponse authResponse = authRequestResult.Response())
+{
+    //To obtain the authorization code
+    //authResponse.Code();
+
+    //To obtain the access token
+    DoTokenExchange(authResponse);
+}
+else
+{
+    AuthFailure authFailure = authRequestResult.Failure();
```

Comment:
The "things with extended errors" pattern (I need to publish our guidelines someday - see https://www.osgwiki.com/wiki/WinRT_API_Design_Patterns#Extended_Errors for internal folks) says that this should be like:

```c++
runtimeclass OAuth2Response
{
    String Token { get; }; // The token
    OAuth2ResponseStatus Status { get; }; // The response code, from the known set of OAuth2 responses
    HRESULT ExtendedError { get; };
    String AdditionalContext { get; }; // Any other textual content the server responded with
}
```